lua/mapvote/client/modules/thumb_downloader.lua

```diff
@@ -68,31 +66,37 @@ end
 function ThumbDownloader:RequestWorkshopIDs()
     if #self.mapsToDownload == 0 then return end
 
-    MapVote.Net.requestWorskhopIDs( self.mapsToDownload )
+    local mapsNeedingIDs = {}
+    for _, mapData in pairs( self.mapsToDownload ) do
+        local map = mapData.map
+        if not self.workshopIDLookup[map] and not self.urlOverrides[map] then
```

Comment:
```suggestion
        if not self.urlOverrides[map] then
```
Because requesting workshop IDs results in a full overwrite of the table I think you dont want to skip maps already in the table. The current  code will result in missing icons on repeat votes